create-doks-with-terraform-flux/README.md

```diff
@@ -1 +1,470 @@
+
+[Flux CD](https://fluxcd.io) is used for managing the `Continuous Delivery` of applications inside the DigitalOcean Kubernetes cluster as well as a `Gitops` enabler.
+
+
+## Bootstrapping DOKS and Flux CD
+
+This section contains information about how we can bootstrap **DOKS** and **Flux CD** via **Terraform** using **Github** as a SCM provider and source of truth. The bootstrap process will create for us a DOKS cluster alongside with a new Github repository and branch where all the manifests used to provision [Flux CD](https://fluxcd.io) will be stored. The same repository will be used to store our Kubernetes custom application deployment manifests that will be managed automatically via [Flux CD](https://fluxcd.io).
+
+### Requirements
+
+1. A DigitalOcean [account](https://cloud.digitalocean.com) to create the access keys and provision the DOKS cluster.
+2. A [Github](https://github.com/) account.
+3. A git client has to be installed as well depending on the distro.
+  
+    E.g.: on MacOS it can be installed via Homebrew:
+    ```bash
+    brew info git
+    brew install git
+    ```
+
+4. [Terraform](https://www.terraform.io/downloads.html) must be installed next. 
+    
+    E.g.: on MacOS it can be installed via Homebrew:
+    ```bash
+    brew info terraform
+    brew install terraform
+    ```
+
+### Installation steps
+
+1. Start by creating a [Github personal access token](https://github.com/settings/tokens) that has the `repo` permissions set. The Terraform module provided by this blueprint will automatically create for us a Github repository and branch where all the required Kubernetes manifests will be stored. In order to achieve this we must create a few special environment variables so make sure to replace the `<>` placeholders accordingly:
+
+    ```bash
+    export TF_VAR_github_owner="<github_username>"                              # should be pretty self explanatory
+    export TF_VAR_github_token="<github_personal_access_token>"                 # should be pretty self explanatory
+    export TF_VAR_github_repository_name="<git_repository_name>"                # the desired git repository name
+    export TF_VAR_github_repository_branch="<git_repository_branch>"            # branch name to be created
+    export TF_VAR_github_repository_target_path="<flux_cd_sync_target_path>"    # the K8S manifests used to provision Flux CD will be stored in this path
+    ```
+2. Let's go next with setting up the required stuff on the [DigitalOcean](https://cloud.digitalocean.com) side by creating the necessary tokens first. One is needed for creating/managing the DOKS cluster and another one for [DO Spaces](https://cloud.digitalocean.com/spaces) (S3-compatibe object storage). The latter is needed for storing the Terraform state file.
```

Comment:
```suggestion
2. Create a DigitalOcean access token for creating/managing the DOKS cluster and another one for [DigitalOcean Spaces](https://cloud.digitalocean.com/spaces) (S3-compatible object storage). We use Spaces for storing the Terraform state file.
```

Reply:
Thanks for the suggestion. Fixed in the latest commit.